src/transaction.rs

```diff
@@ -0,0 +1,254 @@
+use crate::{
+    ColumnFamily, CreateIter, DBRawIterator, DBVector, Error, OptimisticTransactionDB, ReadOptions,
+};
+use ffi;
+use libc::{c_char, size_t};
+use std::marker::PhantomData;
+
+pub struct Transaction<'a> {
+    pub(crate) inner: *mut ffi::rocksdb_transaction_t,
+    pub(crate) db: PhantomData<&'a OptimisticTransactionDB>,
+    snapshot: Option<*const ffi::rocksdb_snapshot_t>,
+}
+
+impl<'a> Transaction<'a> {
+    pub(crate) fn new(
```

Comment:
Why does this need to be `pub(crate)`?

Reply:
visit it from src/optimistic_transaction_db.rs and src/transaction_db.rs